lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,6 +38,18 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not isinstance(text, str) \
+            or not (isinstance(start_of_word, str) or start_of_word is None) \
+            or not isinstance(end_of_word, str):
+        return None
+    freq_dict = {}
+    words_list = text.split()
+    for word in words_list:
+        tokens_tuple = prepare_word(word, start_of_word, end_of_word)
+        if tokens_tuple is None:
+            return None
+        freq_dict.update({tokens_tuple: words_list.count(word)})
```

Comment:
Isn't `update` a bit of an overkill here? Consider adding a key to the dictionary in a traditional way, and it would be good to check if the key in the dictionary first

Reply:
I was thinking about it too. However, I couldn’t realise it before checking. Nevertheless, I’ve accomplished it tonight. So I’ll have pushed these changes by the next check on Thursday.